### DISCUSSION TITLE
Consider sealing the deployment automatically as a “cleanup” step

### DISCUSSION THREAD

**Discussion by mxcl** [bobheadxi/deployments#38] [Ideas]
GitHub Actions support a “cleanup” step for all actions.

This action could use that to automatically seal the deployment, using the failure or success status of the job to determine what is done.

If possible, I’d prefer this since I use this action in many places and having to specify it twice makes it take a lot of “cognitive space” in my workflows.

**Comment by bobheadxi** [comment]
wow, it even seems to have `post-if`! I don't think this existed when I first created this action - I'll definitely look into it. Thanks for pointing this out! Added to #36 

For reference: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#post

**Reply by bobheadxi** [reply]
I can't seem to find any documentation on whether `{{ job.status }} is available to these `post` steps - I can't use `post-if` because I need to account for both success (set to "done") and failure (set to "failed") cases. You can't `{{ job.status }}` at the start since that won't reflect the end state, so unfortunately this might not be possible :/

If anyone has ideas on this front let me know!
